recipe/meta.yaml

```diff
@@ -25,15 +16,12 @@ requirements:
     - cmake   >=3.16.3
     - ninja
     - lua     5.4.*
-    - swig    4.0.*
     - {{ compiler('c') }}
+    - {{ stdlib("c") }}
     - {{ compiler('cxx') }}
 
   host:
-    - libitk-devel  5.3.*
-
-  run:
-    - {{ pin_compatible('libitk', exact=True) }}
+    - libitk-devel  5.4.*
```

Comment:
I understand that this project is likely closely tied to its. But this kind of stuff breaks the global pinnings we have and the tooling we use at conda forge. 

Sometimes it is necessary to use it (I use it pyside and qt) but I just want to make sure you know that this package should be the exception and not the rules for this kind of syntax. 

Downstream libraries should ideally be compatible with a small range of versions of itk.

Reply:
To be explicit about what happens when you add this manual pin. The migrator will stop issuing PR and the pinning will be removed from the ci support files

Reply:
Firstly thank you for your work on these packages, it is very much appreciated and has been needed.

Yes, pyside and Qt would be a reasonable comparison.

>To be explicit about what happens when you add this manual pin. The migrator will stop issuing PR and the pinning will be removed from the ci support files

The intention here is that libsimpleitk should only be built against the specified ITK Major.Minor version. Also given that ITK is a C++ template library, when patch releases are made to ITK, a rebuild of dependent packages is likely needed as the "patches" in ITK are likely in the uncompiled template code of the library.


Also this package is also a "problematic"  condo forge package, as it actually is just a "dev" static library that is only used by the [simpleitk-feedstock](https://github.com/conda-forge/simpleitk-feedstock). This dev package is compiled once per platform and takes 2hr+ and is very large because it exposes on the ITK C++ template symbols.
 
The python SimpleITK-feedstock is then recompiled with this "dev" library for each OS, python version matrixed. This is done in ~10 min, and resulting packages are 10x smaller that if shared libraries were used ( 100s of MB of template symbols in the shared library interfaces).

I would think that ideally one feed stock would build the libsimpleITK per platform, and then use that artifact for each of combination in the python matrix. But as it is now the pinning and rebuilding processes do not work for the dependencies of  SimpleITK-feedstock.

Reply:
Re 2hr + 10min builds:
maybe we can take inspiration from the megabuild by Isuru for pytorch that has a similar problem:
https://github.com/conda-forge/pytorch-cpu-feedstock/pull/210